frontend/src/components/cluster/KubeConfigLoader.tsx

```diff
@@ -145,10 +145,12 @@ function KubeConfigLoader() {
         const selectedClusterConfig = configWithSelectedClusters(fileContent, selectedClusters);
         setCluster({ kubeconfig: btoa(yaml.dump(selectedClusterConfig)) })
           .then(res => {
-            if (res.clusters.length > 0) {
+            // @todo: The setCluster may return a response or not. This is not consistent, so we
+            // should make sure we always either have a response or we don't.
+            if (res?.clusters?.length > 0) {
               dispatch(setConfig(res));
-              setState(Step.Success);
             }
+            setState(Step.Success);
```

Comment:
Since kubeconfig is sent in the request. We are considering it as stateless cluster. So we should use `setStatelessConfig` here. Tested it

Reply:
Made the push